featurebyte/routes/base_router.py

```diff
@@ -0,0 +1,204 @@
+"""
+Base router
+"""
+from typing import Generic, Optional, Type, TypeVar, Union, cast
+
+from http import HTTPStatus
+
+from fastapi import APIRouter, Request
+
+from featurebyte.models.base import PydanticObjectId
+from featurebyte.models.persistent import AuditDocumentList
+from featurebyte.routes.common.schema import (
+    AuditLogSortByQuery,
+    NameQuery,
+    PageQuery,
+    PageSizeQuery,
+    SearchQuery,
+    SortByQuery,
+    SortDirQuery,
+)
+from featurebyte.routes.lazy_app_container import LazyAppContainer
+from featurebyte.schema.common.base import DeleteResponse, DescriptionUpdate
+
+ObjectModelT = TypeVar("ObjectModelT")
+ListObjectModelT = TypeVar("ListObjectModelT")
+CreateObjectSchemaT = TypeVar("CreateObjectSchemaT")
+ControllerT = TypeVar("ControllerT")
+
+
+class BaseRouter:
+    """
+    BaseRouter class that just encapsulates the APIRouter object.
+    """
+
+    def __init__(self, router: APIRouter) -> None:
+        self.router = router
+
+
+class BaseApiRouter(
+    BaseRouter, Generic[ObjectModelT, ListObjectModelT, CreateObjectSchemaT, ControllerT]
+):
+    """
+    Base API router.
+
+    This class contains basic CRUD operations for a given model.
+    """
+
+    object_model: Type[ObjectModelT]
+    list_object_model: Type[ListObjectModelT]
+    create_object_schema: Type[CreateObjectSchemaT]
+    controller: Union[Type[ControllerT], str]
+
+    def __init__(self, prefix: str, skip_id_prefix: bool = False):
+        super().__init__(router=APIRouter(prefix=prefix))
+        base_name = prefix.lstrip("/")
+        api_id = f"{{{base_name}_id}}" if not skip_id_prefix else "{object_id}"
```

Comment:
we need this to maintain backwards compatibility with the api route names. going forward, we can standardize the id params to `object_id`